File: frontend/src/types/team.ts
export type Continent = '亚洲' | '欧洲' | '南美洲' | '北美洲' | '非洲' | '大洋洲'

/** Backend raw team */
export interface ApiTeam {
  id: number
  name: string
  name_en: string
  fifa_code: string | null
  flag_url: string
  continent: string
  group_id?: number | null
  group: { id: number; name: string } | null
  description: string
  coach: string
  external_code?: string | null
  team_type?: string
  country?: string
  venue?: string
}

/** Frontend normalized team */
export interface Team {
  id: number
  name: string
  name_en: string
  code: string
  flag: string
  group_id?: number | null
  group_name: string
  continent: Continent
  team_type?: string
  country?: string
  venue?: string
  is_followed?: boolean
}

export function normalizeTeam(t: ApiTeam): Team {
  return {
    id: t.id,
    name: t.name,
    name_en: t.name_en,
    code: t.fifa_code || t.external_code || '',
    flag: t.flag_url,
    group_id: t.group_id,
    group_name: t.group?.name || '',
    continent: (t.continent as Continent) || '欧洲',
    team_type: t.team_type,
    country: t.country,
    venue: t.venue,
  }
}

File: backend/internal/models/team.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Team struct {
	ID           uint           `gorm:"primaryKey" json:"id"`
	Name         string         `gorm:"size:50;not null;index" json:"name"`
	NameEn       string         `gorm:"size:50;index" json:"name_en"`
	FIFACode     *string        `gorm:"size:10;uniqueIndex" json:"fifa_code"`            // national teams only; NULL for clubs (column stays nullable, unique index untouched)
	ExternalCode *string        `gorm:"size:20;index" json:"external_code"`              // provider code (e.g. football-data tla) for clubs
	TeamType     string         `gorm:"size:20;default:national;index" json:"team_type"` // national | club
	FlagURL      string         `gorm:"size:255" json:"flag_url"`
	Continent    string         `gorm:"size:20;index" json:"continent"`
	Country      string         `gorm:"size:50" json:"country"` // club country
	Venue        string         `gorm:"size:100" json:"venue"`  // club home stadium
	GroupID      *uint          `gorm:"index" json:"group_id"`  // national teams only; NULL for clubs
	Group        Group          `gorm:"foreignKey:GroupID" json:"group,omitempty"`
	Description  string         `gorm:"type:text" json:"description"`
	Coach        string         `gorm:"size:100" json:"coach"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `gorm:"index" json:"-"`
}
